nixos/modules/services/networking/nat.nix

```diff
@@ -319,7 +319,7 @@ in
         }
       ];
 
-      environment.systemPackages = [ pkgs.iptables ];
+      environment.systemPackages = optional (!config.networking.firewall.enable) config.networking.firewall.package;
```

Comment:
I'm confused, you add the firewall package only when the firewall is disabled?

Reply:
It's added already by the firewall module if `networking.firewall.enable` is true. Should I always add it regardless?

Reply:
If `firewall.enable` already adds `config.networking.firewall.package` to `systemPackages`, then you can probably just remove this line altogether.

Reply:
The intention is to have iptables in systemPackages (the existing behavior) even when firewall is _not_ enabled.

Reply:
That seems like weird behavior, cc. @andir @flokli

Reply:
I'll take a look at this tomorrow.

Reply:
Sorry, took a bit longer, it took me a while to grok this :-)

`nat.nix` works like this: in case the firewall is enabled, it sets some `extra[Stop]Commands` in `networking.firewall` containing iptables invocations, otherwise, it registers a new standalone systemd service, running the same scripts only.

Apparently `config.networking.firewall.package` is not used only for the firewall, but for the nat module, too, even if the firewall is disabled. It's not pretty, but that's what we're stuck for now I guess.

I think we can just set
```
environment.systemPackages = [ config.networking.firewall.package ];
```

in `nat.nix` in case nat is enabled. In case the firewall is also enabled, both modules add the same identical package to `environment.systemPackages`, which should just deduplicate without any error, and is probably less confusing.

We should still add a comment next to the line explaining this.

Reply:
Thanks a bunch flokli, that makes a _lot_ more sense :)